**Desafio 1**

In [2]:
import pandas as pd
import numpy as np


In [3]:
df = pd.read_csv("worldcup2014.csv")

In [4]:
df.head()

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

**Desafio 2**

In [5]:
df["continent"].value_counts()

europe          13
southamerica     6
africa           5
asia             4
northamerica     4
Name: continent, dtype: int64

¿Cuál es el continente con una mayor presencia en la muestra?

In [6]:
(df["continent"].value_counts('%'))*100

europe          40.625
southamerica    18.750
africa          15.625
asia            12.500
northamerica    12.500
Name: continent, dtype: float64

Segun lo observado en la tabla de arriba, Europa tiene la mayor presencia en la muestra con un 40,625%

¿Cuál es la probabilidad de elegir un equipo asiático al azar? 

In [7]:
df['continent'].value_counts().sum()

32

In [8]:
df['continent'].value_counts()['asia']

4

In [9]:
(df['continent'].value_counts()['asia']/df['continent'].value_counts().sum())*100

12.5

La probabilidad de elegir un equipo asiatico al azar es de 12.5%

¿Cuál es la probabilidad de elegir un equipo africano al azar? 

In [10]:
(df['continent'].value_counts()['africa']/df['continent'].value_counts().sum())*100

15.625

La probabilidad de elegir un equipo africano al azar es de un 15,6%

**Desafio 3**

In [11]:
df_asia = df[df['continent'] == 'asia'] 
df_asia

team continent group  group_pos  cantidad_juegos  juegos_ganados  \
7     Australia      asia     B          4                3               0   
11        Japan      asia     C          4                3               0   
23         Iran      asia     F          4                3               0   
31  South Korea      asia     H          4                3               0   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
7                  0                3            3             9   
11                 1                2            2             6   
23                 1                2            1             4   
31                 1                2            3             6   

   goles_diferencia  puntos  clasificado  
7                −6       0            0  
11               −4       1            0  
23               −3       1            0  
31               −3       1            0

In [12]:
df_africa = df[df['continent'] == 'africa']
df_africa

team continent group  group_pos  cantidad_juegos  juegos_ganados  \
3      Cameroon    africa     A          4                3               0   
10  Ivory Coast    africa     C          3                3               1   
21      Nigeria    africa     F          2                3               1   
27        Ghana    africa     G          4                3               0   
29      Algeria    africa     H          2                3               1   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
3                  0                3            1             9   
10                 0                2            4             5   
21                 1                1            3             3   
27                 1                2            4             6   
29                 1                1            6             5   

   goles_diferencia  puntos  clasificado  
3                −8       0            0  
10               −1       3            0  
21                0       4            1  
27               −2       1            0  
29                1       4            1

In [13]:
df_northamerica = df[df['continent'] == 'northamerica']
df_northamerica

team     continent group  group_pos  cantidad_juegos  \
1          Mexico  northamerica     A          2                3   
12     Costa Rica  northamerica     D          1                3   
19       Honduras  northamerica     E          4                3   
25  United States  northamerica     G          2                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
1                2                 1                0            4   
12               2                 1                0            4   
19               0                 0                3            1   
25               1                 1                1            4   

    goles_contra goles_diferencia  puntos  clasificado  
1              1                3       7            1  
12             1                3       7            1  
19             8               −7       0            0  
25             4                0       4            1

In [14]:
df_europe = df[df['continent'] == 'europe']
df_europe

team continent group  group_pos  cantidad_juegos  \
2                  Croatia    europe     A          3                3   
4              Netherlands    europe     B          1                3   
6                    Spain    europe     B          3                3   
9                   Greece    europe     C          2                3   
14                   Italy    europe     D          3                3   
15                 England    europe     D          4                3   
16                  France    europe     E          1                3   
17             Switzerland    europe     E          2                3   
22  Bosnia and Herzegovina    europe     F          3                3   
24                 Germany    europe     G          1                3   
26                Portugal    europe     G          3                3   
28                 Belgium    europe     H          1                3   
30                  Russia    europe     H          3                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
2                1                 0                2            6   
4                3                 0                0           10   
6                1                 0                2            4   
9                1                 1                1            2   
14               1                 0                2            2   
15               0                 1                2            2   
16               2                 1                0            8   
17               2                 0                1            7   
22               1                 0                2            4   
24               2                 1                0            7   
26               1                 1                1            4   
28               3                 0                0            4   
30               0                 2                1            2   

    goles_contra goles_diferencia  puntos  clasificado  
2              6                0       3            0  
4              3                7       9            1  
6              7               −3       3            0  
9              4               −2       4            1  
14             3               −1       3            0  
15             4               −2       1            0  
16             2                6       7            1  
17             6                1       6            1  
22             4                0       3            0  
24             2                5       7            1  
26             7               −3       4            0  
28             1                3       9            1  
30             3               −1       2            0

In [15]:
df_southamerica = df[df['continent'] == 'southamerica']
df_southamerica

team     continent group  group_pos  cantidad_juegos  juegos_ganados  \
0      Brazil  southamerica     A          1                3               2   
5       Chile  southamerica     B          2                3               2   
8    Colombia  southamerica     C          1                3               3   
13    Uruguay  southamerica     D          2                3               2   
18    Ecuador  southamerica     E          3                3               1   
20  Argentina  southamerica     F          1                3               3   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
0                  1                0            7             2   
5                  0                1            5             3   
8                  0                0            9             2   
13                 0                1            4             4   
18                 1                1            3             3   
20                 0                0            6             3   

   goles_diferencia  puntos  clasificado  
0                 5       7            1  
5                 2       6            1  
8                 7       9            1  
13                0       6            1  
18                0       4            0  
20                3       9            1

**Desafio 4**


In [16]:
clasificado_por_continent = df.groupby(['continent']).mean()
clasificado_por_continent

group_pos  cantidad_juegos  juegos_ganados  juegos_empatados  \
continent                                                                    
africa         3.000000              3.0        0.600000          0.600000   
asia           4.000000              3.0        0.000000          0.750000   
europe         2.307692              3.0        1.384615          0.538462   
northamerica   2.250000              3.0        1.250000          0.750000   
southamerica   1.666667              3.0        2.166667          0.333333   

              juegos_perdidos  goles_favor  goles_contra    puntos  \
continent                                                            
africa               1.800000     3.600000      5.600000  2.400000   
asia                 2.250000     2.250000      6.250000  0.750000   
europe               1.076923     4.769231      4.000000  4.692308   
northamerica         1.000000     3.250000      3.500000  4.500000   
southamerica         0.500000     5.666667      2.833333  6.833333   

              clasificado  
continent                  
africa           0.400000  
asia             0.000000  
europe           0.461538  
northamerica     0.750000  
southamerica     0.833333

In [17]:
clasificado_por_continent['clasificado']*100

continent
africa          40.000000
asia             0.000000
europe          46.153846
northamerica    75.000000
southamerica    83.333333
Name: clasificado, dtype: float64

En la tabla anterior se puede ver la probabilidad de que un pais clasifique por continente.


Un pais de Africa tiene un 40% de probabilidades de clasificar.

Un pais de Asia tiene un 0% de probabilidades de clasificar.

Un pais de Europa tiene un 46% de probabilidades de clasificar.

Un pais de Northamerica tiene un 75% de probabilidades de clasificar.

Un pais de Southamerica tiene un 83% de probabilidades de clasificar.

***¿Cuál es la probabilidad de que un país asiático pase a la siguiente ronda?***


La probabilidad es de 0%

***¿Cuáles son los dos continentes con la mayor probabilidad de clasificar?***

Southamerica y Northamerica

***¿Cuál es la probabilidad de que un país europeo no clasifique?***

In [18]:
(1-clasificado_por_continent['clasificado']['europe'])*100 

53.84615384615385

La probabilidad de que un pais europeo no clasifique es: 53.84%

**Desafio 5**

In [19]:
df['ha_ganado'] = np.where(df['juegos_ganados'] == 0, 0, 1) 

In [20]:
df

team     continent group  group_pos  cantidad_juegos  \
0                   Brazil  southamerica     A          1                3   
1                   Mexico  northamerica     A          2                3   
2                  Croatia        europe     A          3                3   
3                 Cameroon        africa     A          4                3   
4              Netherlands        europe     B          1                3   
5                    Chile  southamerica     B          2                3   
6                    Spain        europe     B          3                3   
7                Australia          asia     B          4                3   
8                 Colombia  southamerica     C          1                3   
9                   Greece        europe     C          2                3   
10             Ivory Coast        africa     C          3                3   
11                   Japan          asia     C          4                3   
12              Costa Rica  northamerica     D          1                3   
13                 Uruguay  southamerica     D          2                3   
14                   Italy        europe     D          3                3   
15                 England        europe     D          4                3   
16                  France        europe     E          1                3   
17             Switzerland        europe     E          2                3   
18                 Ecuador  southamerica     E          3                3   
19                Honduras  northamerica     E          4                3   
20               Argentina  southamerica     F          1                3   
21                 Nigeria        africa     F          2                3   
22  Bosnia and Herzegovina        europe     F          3                3   
23                    Iran          asia     F          4                3   
24                 Germany        europe     G          1                3   
25           United States  northamerica     G          2                3   
26                Portugal        europe     G          3                3   
27                   Ghana        africa     G          4                3   
28                 Belgium        europe     H          1                3   
29                 Algeria        africa     H          2                3   
30                  Russia        europe     H          3                3   
31             South Korea          asia     H          4                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0                2                 1                0            7   
1                2                 1                0            4   
2                1                 0                2            6   
3                0                 0                3            1   
4                3                 0                0           10   
5                2                 0                1            5   
6                1                 0                2            4   
7                0                 0                3            3   
8                3                 0                0            9   
9                1                 1                1            2   
10               1                 0                2            4   
11               0                 1                2            2   
12               2                 1                0            4   
13               2                 0                1            4   
14               1                 0                2            2   
15               0                 1                2            2   
16               2                 1                0            8   
17               2                 0                1            7   
18               1                 1                1            3   
19               0                 0           

In [22]:
df.groupby(['continent']).mean()

group_pos  cantidad_juegos  juegos_ganados  juegos_empatados  \
continent                                                                    
africa         3.000000              3.0        0.600000          0.600000   
asia           4.000000              3.0        0.000000          0.750000   
europe         2.307692              3.0        1.384615          0.538462   
northamerica   2.250000              3.0        1.250000          0.750000   
southamerica   1.666667              3.0        2.166667          0.333333   

              juegos_perdidos  goles_favor  goles_contra    puntos  \
continent                                                            
africa               1.800000     3.600000      5.600000  2.400000   
asia                 2.250000     2.250000      6.250000  0.750000   
europe               1.076923     4.769231      4.000000  4.692308   
northamerica         1.000000     3.250000      3.500000  4.500000   
southamerica         0.500000     5.666667      2.833333  6.833333   

              clasificado  ha_ganado  
continent                             
africa           0.400000   0.600000  
asia             0.000000   0.000000  
europe           0.461538   0.846154  
northamerica     0.750000   0.750000  
southamerica     0.833333   1.000000

***¿Qué continente tiene la mayor probabilidad de tener países con al menos 1
victoria?***

Basado en el cuadro de arriba, el continente con mayor probabilidad de tener paises con al menos 1 victoria es Southamerica.

***¿Qué continente tiene un nivel similar entre países que tienen o no tienen
victorias?***

El continente que tiene un porcentaje más cercano al 50%, es africa con un 60%. 

**Reformule el código para juegos perdidos y analice los continentes con mayor
probabilidad de tener países sin juegos perdidos.**

In [28]:
df['ha_perdido'] = np.where(df['juegos_perdidos']==0, 1, 0)

In [29]:
df

team     continent group  group_pos  cantidad_juegos  \
0                   Brazil  southamerica     A          1                3   
1                   Mexico  northamerica     A          2                3   
2                  Croatia        europe     A          3                3   
3                 Cameroon        africa     A          4                3   
4              Netherlands        europe     B          1                3   
5                    Chile  southamerica     B          2                3   
6                    Spain        europe     B          3                3   
7                Australia          asia     B          4                3   
8                 Colombia  southamerica     C          1                3   
9                   Greece        europe     C          2                3   
10             Ivory Coast        africa     C          3                3   
11                   Japan          asia     C          4                3   
12              Costa Rica  northamerica     D          1                3   
13                 Uruguay  southamerica     D          2                3   
14                   Italy        europe     D          3                3   
15                 England        europe     D          4                3   
16                  France        europe     E          1                3   
17             Switzerland        europe     E          2                3   
18                 Ecuador  southamerica     E          3                3   
19                Honduras  northamerica     E          4                3   
20               Argentina  southamerica     F          1                3   
21                 Nigeria        africa     F          2                3   
22  Bosnia and Herzegovina        europe     F          3                3   
23                    Iran          asia     F          4                3   
24                 Germany        europe     G          1                3   
25           United States  northamerica     G          2                3   
26                Portugal        europe     G          3                3   
27                   Ghana        africa     G          4                3   
28                 Belgium        europe     H          1                3   
29                 Algeria        africa     H          2                3   
30                  Russia        europe     H          3                3   
31             South Korea          asia     H          4                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0                2                 1                0            7   
1                2                 1                0            4   
2                1                 0                2            6   
3                0                 0                3            1   
4                3                 0                0           10   
5                2                 0                1            5   
6                1                 0                2            4   
7                0                 0                3            3   
8                3                 0                0            9   
9                1                 1                1            2   
10               1                 0                2            4   
11               0                 1                2            2   
12               2                 1                0            4   
13               2                 0                1            4   
14               1                 0                2            2   
15               0                 1                2            2   
16               2                 1                0            8   
17               2                 0                1            7   
18               1                 1                1            3   
19               0                 0           

In [30]:
df.groupby(['continent']).mean()

group_pos  cantidad_juegos  juegos_ganados  juegos_empatados  \
continent                                                                    
africa         3.000000              3.0        0.600000          0.600000   
asia           4.000000              3.0        0.000000          0.750000   
europe         2.307692              3.0        1.384615          0.538462   
northamerica   2.250000              3.0        1.250000          0.750000   
southamerica   1.666667              3.0        2.166667          0.333333   

              juegos_perdidos  goles_favor  goles_contra    puntos  \
continent                                                            
africa               1.800000     3.600000      5.600000  2.400000   
asia                 2.250000     2.250000      6.250000  0.750000   
europe               1.076923     4.769231      4.000000  4.692308   
northamerica         1.000000     3.250000      3.500000  4.500000   
southamerica         0.500000     5.666667      2.833333  6.833333   

              clasificado  ha_ganado  ha_perdido  
continent                                         
africa           0.400000   0.600000    0.000000  
asia             0.000000   0.000000    0.000000  
europe           0.461538   0.846154    0.307692  
northamerica     0.750000   0.750000    0.500000  
southamerica     0.833333   1.000000    0.500000

In [33]:
df.groupby(['continent']).mean()['ha_perdido']*100

continent
africa           0.000000
asia             0.000000
europe          30.769231
northamerica    50.000000
southamerica    50.000000
Name: ha_perdido, dtype: float64

Basado en la tabla anterior, los continentes con mayor probabilidad de tener paises sin juegos perdidos son Southamerica y Northamerica. 